subsystem/apiclient/kubeapi.go

```diff
@@ -0,0 +1,288 @@
+package apiclient
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/google/uuid"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/internal/common"
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	APIVersion = "adi.io.my.domain/v1alpha1"
+)
+
+type KubeAPIClient struct {
+	client    runtimeclient.Client
+	namespace string
+	db        *gorm.DB
+}
+
+func NewKubeAPIClient(
+	namespace string,
+	config *rest.Config,
+	db *gorm.DB,
+) (*KubeAPIClient, error) {
+
+	client, err := runtimeclient.New(config, runtimeclient.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		return nil, err
+	}
+	kc := &KubeAPIClient{
+		client:    client,
+		namespace: namespace,
+		db:        db,
+	}
+	return kc, nil
+}
+
+func (kc *KubeAPIClient) RegisterCluster(
+	ctx context.Context,
+	params *installer.RegisterClusterParams,
+) (*models.Cluster, error) {
+
+	cParams := params.NewClusterParams
+
+	clusterName := swag.StringValue(cParams.Name)
+
+	pullSecretRef, pullSecretErr := kc.deployPullSecret(
+		ctx,
+		clusterName,
+		swag.StringValue(cParams.PullSecret))
+	if pullSecretErr != nil {
+		return nil, errors.Wrapf(pullSecretErr, "failed to deploy pull secret")
+	}
+
+	metaName := uuid.New().String()
```

Comment:
Should this just be "clusterName"? We are most likely targeting one cluster per namespace, not much risk of collision. But as with previous comment I'm slightly confused what this layer is as I'd expect this Kube CR to be provided by a user, and most likely with a more friendly meta name.